testenv/utils/ssh.go

```diff
@@ -225,6 +225,25 @@ func (f *SSHPortForward) Stop() {
 	f.cancel()
 }
 
+func (f *SSHPortForward) DialWithTimeout() (*ssh.Client, error) {
```

Comment:
we should handle context cancellation from parent here to avoid waiting full timeout
```suggestion
func (f *SSHPortForward) DialWithTimeout(ctx context.Context) (*ssh.Client, error) {
```